action/protocol/staking/vote_reviser.go

```diff
@@ -35,14 +37,48 @@ func (vr *VoteReviser) Revise(csm CandidateStateManager, height uint64) error {
 		if err != nil {
 			return err
 		}
+		sort.Sort(cands)
+		if vr.needCorrectCands(height) {
+			cands, err = vr.correctAliasCands(csm, cands)
+			if err != nil {
+				return err
+			}
+		}
 		vr.storeToCache(height, cands)
 	}
 	return vr.flush(height, csm)
 }
 
+func (vr *VoteReviser) correctAliasCands(csm CandidateStateManager, cands CandidateList) (CandidateList, error) {
+	var retval CandidateList
+	for _, c := range csm.DirtyView().candCenter.base.nameMap {
+		retval = append(retval, c)
+	}
+	for _, c := range csm.DirtyView().candCenter.base.operatorMap {
+		retval = append(retval, c)
+	}
```

Comment:
we should not use `name/operatorMap` here? don't quite understand the code.

here's my understanding of the intended logic per our discussion during v1.8.4 fix
input `cands` is the list of candidates (with vote correctly revised) in the statedb, as done in v1.8.4 fix, we saved the list of candidates that have been called `CandidateRegister/Update` actions during the affected period, these are candidates that can potentially have an alias in the candidate center
```
_, _, owners := readCandCenterStateFromStateDB()
```
so on top of `cands`, update the name/operator/reward of `owners`, that is what we need to do

Reply:
another comment is that `VoteReviser` logic is clean, it just reads candidates/buckets from statedb, and tally votes again.

now it needs to deal with `csm.DirtyView().candCenter.xxx`, this adds unnecessary code coupling

Reply:
https://github.com/iotexproject/iotex-core/blob/e0578dacb9752611a4a94c4efe4774f7ece0f320/action/protocol/staking/protocol.go#L182 has called and loaded the data.
https://github.com/iotexproject/iotex-core/blob/ee04a051811a9c23b33a291fb66c2de957b78bf3/action/protocol/staking/handlers.go#L689 candcenter has been used in various places, which is a legacy issue

Reply:
yes, ideally we should not use `csm.DirtyView().candCenter.xxx` in `protocol.go` and `handlers.go`
so we should not expand it into `vote_reviser.go` if possible

Reply:
`name, operator, owners, err := readCandCenterStateFromStateDB(sr, height)`, will NOT be called at next HF height